integration-tests/chain-signatures/tests/actions/mod.rs

```diff
@@ -51,39 +50,24 @@ pub async fn request_sign(
         public_key: account.secret_key().public_key().to_string().parse()?,
         secret_key: account.secret_key().to_string().parse()?,
     };
-    let (nonce, block_hash, _) = ctx
-        .rpc_client
-        .fetch_nonce(&signer.account_id, &signer.public_key)
-        .await?;
 
     let request = SignRequest {
         payload: payload_hashed,
         path: "test".to_string(),
         key_version: 0,
     };
-    let tx_hash = ctx
-        .jsonrpc_client
-        .call(&RpcBroadcastTxAsyncRequest {
-            signed_transaction: Transaction {
-                nonce,
-                block_hash,
-                signer_id: signer.account_id.clone(),
-                public_key: signer.public_key.clone(),
-                receiver_id: ctx.nodes.ctx().mpc_contract.id().clone(),
-                actions: vec![Action::FunctionCall(Box::new(FunctionCallAction {
-                    method_name: "sign".to_string(),
-                    args: serde_json::to_vec(&serde_json::json!({
-                        "request": request,
-                    }))?,
-                    gas: 300_000_000_000_000,
-                    deposit: 1,
-                }))],
-            }
-            .sign(&signer),
-        })
+    let status = ctx
+        .rpc_client
+        .call(&signer, ctx.nodes.ctx().mpc_contract.id(), "sign")
+        .args_json(serde_json::json!({
+            "request": request,
+        }))
+        .max_gas()
+        .deposit(NearToken::from_yoctonear(1))
+        .transact_async()
```

Comment:
nice enhancement, i've seen concurrent tests fail due to invalid nonce

Reply:
yeah hopefully this resolves that a bit. It won't retry though since near-fetch requires explicit `.retry_expoential(...)`